sonar-kotlin-plugin/src/test/java/org/sonarsource/kotlin/plugin/KotlinSensorTest.kt

```diff
@@ -529,6 +533,31 @@ internal class KotlinSensorTest : AbstractSensorTest() {
         verify(exactly = 1) { context.canSkipUnchangedFiles() }
     }
 
+    @Test
+    fun `test sensor skips cached files`() {
+        val messageDigest = MessageDigest.getInstance("MD5")
+        val files = incrementalAnalysisFileSet()
+        val fileHash = messageDigest.digest(files[InputFile.Status.CHANGED]!!.contents().byteInputStream().readAllBytes())
+
+        val readCache = DummyReadCache(mapOf("kotlin:contentHash:MD5:moduleKey:changed.kt" to fileHash))
+        val writeCache = DummyWriteCache(readCache = readCache)
+        context.setNextCache(writeCache)
+        context.setPreviousCache(readCache)
+        context.setCanSkipUnchangedFiles(true)
+        context.isCacheEnabled = true
+
+        files.values.forEach { context.fileSystem().add(it) }
+        val checkFactory = checkFactory("S1764")
+
+        sensor(checkFactory).execute(context)
+        assertThat(logTester.logs(LoggerLevel.DEBUG))
+            .contains("Content hash cache was initialized")
+        assertThat(logTester.logs(LoggerLevel.INFO))
+            .contains("Only analyzing 2 changed Kotlin files out of 3.")
+
+
+    }
```

Comment:
Should we also test a case where we explicitly test when cache is disabled and we fall back on the input file status?